specification/storage/resource-manager/Microsoft.Storage/stable/2021-01-01/file.json

```diff
@@ -0,0 +1,886 @@
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- The Share is successfully restored."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "FileServiceItems": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/FileServiceProperties"
+          },
+          "description": "List of file services returned.",
+          "readOnly": true
+        }
+      }
+    },
+    "FileServiceProperties": {
+      "properties": {
+        "properties": {
+          "properties": {
+            "cors": {
+              "$ref": "./common.json#/definitions/CorsRules",
+              "description": "Specifies CORS rules for the File service. You can include up to five CorsRule elements in the request. If no CorsRule elements are included in the request body, all CORS rules will be deleted, and CORS will be disabled for the File service."
+            },
+            "shareDeleteRetentionPolicy": {
+              "$ref": "./common.json#/definitions/DeleteRetentionPolicy",
+              "description": "The file service properties for share soft delete."
+            },
+            "protocolSettings": {
+              "$ref": "#/definitions/ProtocolSettings",
+              "description": "Protocol settings for file service"
+            }
+          },
+          "x-ms-client-flatten": true,
+          "x-ms-client-name": "FileServiceProperties",
+          "description": "The properties of File services in storage account."
+        },
+        "sku": {
+          "$ref": "./common.json#/definitions/Sku",
+          "readOnly": true,
+          "description": "Sku name and tier."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/Resource"
+        }
+      ],
+      "description": "The properties of File services in storage account."
+    },
+    "ProtocolSettings": {
+      "properties": {
+        "smb": {
+          "$ref": "#/definitions/SmbSetting",
+          "description": "Setting for SMB protocol"
+        }
+      },
+      "description": "Protocol settings for file service"
+    },
+    "SmbSetting": {
+      "properties": {
+        "multichannel": {
+          "$ref": "#/definitions/Multichannel",
+          "description": "Multichannel setting. Applies to Premium FileStorage only."
+        },
+        "versions": {
```

Comment:
>versions [](start = 9, length = 8)

Have you considered modeling these properties as arrays of strings instead of relying on delimiters?

Reply:
This is to align with the other file share options on the storage account, they are also taken as delimited strings today and he mentioned it was to keep the interface consistent.